src/irmin-git/irmin_git.ml

```diff
@@ -717,6 +722,20 @@ functor
           >|= fun () -> b)
 
     let close _ = Lwt.return_unit
+
+    let clear t =
+      Log.debug (fun l -> l "clear");
+      Lwt_mutex.with_lock t.m (fun () ->
+          G.Ref.list t.t >>= fun refs ->
+          Lwt_list.iter_p
+            (fun (r, _) ->
+              G.Ref.remove t.t r >>= function
```

Comment:
This is just removing the reference in the store, not touching the objects at all. It probably doesn't matter much as there is no GC yet in ocaml-git. @dinosaure is there anything else better to do here?

Reply:
Currently, it's not the case, you probably should open an issue into mirage/ocaml-git.

Reply:
I'm not understanding the git backend very well, but this is the reference store, isn't it supposed to only clear the references? In `Content_addressable` we use [reset](https://github.com/mirage/ocaml-git/blob/master/src/git/minimal.ml#L194), which clears everything.

Reply:
Depends on what you want. `G.Ref.remove` will removes only the file which contains the reference. Then, any objects reachable from this reference will still physically exist - and can be considered as garbage. But, as @samoht said, it should not be a blocker for you and, as I said, you probably create an issue on that to track it easily then :+1:.

Reply:
Thank you @dinosaure for clarifying this. 

The question of what we want to clear is not specific to `irmin-git`: in every backend we have a `clear` function for the branch store, and one for the commit store etc. 

I think we want `clear` to only touch the store it is called on. For instance, a commit points to a node, but calling clear on commits does not touch the nodes. 
Moreover, clear should be fast. If we ask to delete everything "under" a reference, then it means some traversal of the graph (vs what clear does now in most backends, which is to reset a hashtable or a file offset).